src/openvic-simulation/economy/GoodDefinition.cpp

```diff
@@ -88,29 +88,46 @@ bool GoodDefinitionManager::load_goods_file(ast::NodeCPtr root) {
 }
 
 bool GoodDefinitionManager::generate_modifiers(ModifierManager& modifier_manager) const {
+	using enum ModifierEffect::format_t;
+
 	bool ret = true;
 
-	const auto good_modifier = [this, &modifier_manager, &ret](std::string_view name, bool is_positive_good) -> void {
+	const auto good_modifier = [this, &modifier_manager, &ret](
+		std::string_view name, bool is_positive_good, auto make_localisation_suffix
+	) -> void {
 		ret &= modifier_manager.register_complex_modifier(name);
+
 		for (GoodDefinition const& good : get_good_definitions()) {
 			ret &= modifier_manager.add_modifier_effect(
-				ModifierManager::get_flat_identifier(name, good.get_identifier()), is_positive_good
+				ModifierManager::get_flat_identifier(name, good.get_identifier()), is_positive_good, PROPORTION_DECIMAL,
+				make_localisation_suffix(good.get_identifier())
 			);
 		}
 	};
 
-	good_modifier("artisan_goods_input", false);
-	good_modifier("artisan_goods_output", true);
-	good_modifier("artisan_goods_throughput", true);
-	good_modifier("factory_goods_input", false);
-	good_modifier("factory_goods_output", true);
-	good_modifier("factory_goods_throughput", true);
-	good_modifier("rgo_goods_output", true);
-	good_modifier("rgo_goods_throughput", true);
-	good_modifier("rgo_size", true);
+	const auto make_production_localisation_suffix = [](std::string_view localisation_suffix) -> auto {
```

Comment:
This doesn't distinguish between factory, rgo or artisan goods modifier.

Reply:
Vic2 doesn't distinguish between them, the tooltips all say `<good> production <input/output/throughput>` regardless of whether it's affecting factory, rgo or artisan production

Reply:
Yes, but does the game still distinguish them internally? I believe that it does

Reply:
We do distinguish them internally, this is just the tooltips he's talking about

Reply:
Confusing. We should support different tooltips. Perhaps we can fallback to the generalised one?

Reply:
We could for custom mode, but for compat mode I don't think it's worth the extra complication this would cause to the modifier localisation system